src/Data/List/Properties.agda

```diff
@@ -1233,15 +1406,46 @@ module _ {x y : A} where
 ∷ʳ-++ : ∀ (xs : List A) (a : A) (ys : List A) → xs ∷ʳ a ++ ys ≡ xs ++ a ∷ ys
 ∷ʳ-++ xs a ys = ++-assoc xs [ a ] ys
 
+------------------------------------------------------------------------
+-- uncons
 
+module _ (f : A → B) where
```

Comment:
Hmm so you've made all the `f`s here and below explicit, but the existing proof `head-map` has an implicit `f`. Can `f` in general be inferable in these situations?

Reply:
right, I tried to figure out what to do from the existing `-map` proofs but they seem to be quite inconsistent, we sometimes use implicit and sometimes explicit. in most of them we do explicit, so that's what I went with for the new `-map` proofs.

as for whether it is in general inferable, it is hard for me to say. I usually prefer the resulting proof to be completely bound by the explicit parameters, but I don't have that much experience in Agda so I'll leave that call to you :)